src/components/Wizard.js

```diff
@@ -82,13 +86,15 @@ class Wizard extends Component {
       if (step.id) {
         this.setState({ step });
       } else {
-        this.history.replace(`${this.basename}${this.ids[0]}`);
+        this.history.replace(`${this.basename}${this.ids[0]}${this.historySuffix}`);
       }
     });
   };
 
-  push = (step = this.nextStep) => this.history.push(`${this.basename}${step}`);
-  replace = (step = this.nextStep) => this.history.replace(`${this.basename}${step}`);
+  push = (step = this.nextStep) =>
```

Comment:
`history.replace` and `history.push` can accept a location-like object which will preserve query params.  check out https://github.com/ReactTraining/history/tree/v4.6.0#navigation.  Using this, you can remove `historySuffix` and do something like this:

```js
replace = (step = this.nextStep) => {
  if (this.props.preserveSearch) {
    this.history.replace({
      ...this.history.location,
      pathname:`${this.basename}${step}`
    })
  } else {
    this.history.replace(`${this.basename}${step}`);
  }
}

```